src/Xrefcheck/System.hs

```diff
@@ -36,21 +46,118 @@ readingSystem = unsafePerformIO
 -- Check the respective env variable which is usually set in all CIs.
 askWithinCI :: IO Bool
 askWithinCI = lookupEnv "CI" <&> \case
-  Just "1"                       -> True
+  Just "1" -> True
   Just (map C.toLower -> "true") -> True
-  _                              -> False
+  _ -> False
+
+-- | A FilePath that has been canonicalized. Should be created via
+-- 'canonicalizePath'. Currently, canonical paths have been made absolute,
+-- normalised regarding the running platform (e.g. Posix or Windows), with
+-- indirections syntractically expanded as much as possible and with no
+-- trailing path separator. All this results in a weaker version than that
+-- provided by 'System.Directory.canonicalizePath'.
+newtype CanonicalPath = UnsafeCanonicalPath FilePath
+  deriving newtype (Show, Eq, Ord)
+
+-- | A 'FilePath' corresponding to the given 'CanonicalPath',
+-- always using POSIX path separators.
+unCanonicalPath :: CanonicalPath -> FilePath
+unCanonicalPath (UnsafeCanonicalPath path) =
+    fmap replaceSeparator path
+  where
+  replaceSeparator c
+    | FP.isPathSeparator c = FPP.pathSeparator
+    | otherwise = c
```

Comment:
Is there any other way to get a `FilePath` from a `CanonicalPath`? If there isn't, why not replace separators on `FilePath` creating?

Reply:
I would prefer to keep the `CanonicalPath` contents with the platform dependant path separator, so they can be used as usual for file system related operations. I'm going to try to move this replacement to a more explicit part.